src/Core/Content/Seo/SeoUrlPlaceholderHandler.php

```diff
@@ -111,6 +126,50 @@ private function createSeoMapping(SalesChannelContext $context, array $mapping):
         return $mapping;
     }
 
+    private function createMediaMapping(SalesChannelContext $context, array $mapping): array
+    {
+        $prefix = '/mediaId/';
+
+        if (empty($mapping)) {
+            return [];
+        }
+
+        $mediaIds = [];
+        foreach ($mapping as $item) {
+            if (str_starts_with($item, $prefix)) {
+                $mediaIds[] = substr($item, \strlen($prefix));
+            }
+        }
+
+        $urls = $this->getMediaUrls($mediaIds, $context->getContext());
+
+        foreach ($urls as $id => $url) {
+            $key = self::DOMAIN_PLACEHOLDER . $prefix . $id . '#';
+            $mapping[$key] = $url;
+        }
+
+        return $mapping;
+    }
+
+    private function getMediaUrls(array $mediaIds, Context $context): array
+    {
+        $criteria = new Criteria($mediaIds);
+        $criteria->setTitle('seo-url-placeholder-handler::get-media-urls');
+        $media = null;
+        $mediaRepository = $this->mediaRepository;
+        $context->scope(Context::SYSTEM_SCOPE, static function (Context $context) use ($criteria, $mediaRepository, &$media): void {
+            /** @var MediaCollection $media */
+            $media = $mediaRepository->search($criteria, $context)->getElements();
+        });
+
+        $urls = [];
+        foreach ($media ?? [] as $mediaItem) {
+            $urls[$mediaItem->id] = $this->urlGenerator->getRelativeMediaUrl($mediaItem);
```

Comment:
Would it work with external filesystem on different domain?
Because "relativeMediaUrl" is not the entire url.

Reply:
Very good catch - I believe it's fixed now.